src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractUnstructuredToArmProcessor.java

```diff
@@ -135,6 +148,9 @@ protected void updateTransferAttempts(ExternalObjectDirectoryEntity externalObje
             externalObjectDirectoryEntity.getId()
         );
         externalObjectDirectoryEntity.setTransferAttempts(currentNumberOfAttempts + 1);
+        if (currentNumberOfAttempts >= armDataManagementConfiguration.getMaxRetryAttempts()) {
+            logApi.armPushFailed(externalObjectDirectoryEntity.getId());
+        }
```

Comment:
Wondering if using `currentNumberOfAttempts` in the conditional here is a bit misleading, because at this point the actual "number of attempts" is actually `currentNumberOfAttempts + 1`.

Would it be clearer to pre-increment in L150, and then use the updated value (and tweak the logic) in the check in L151? Something like:

```java
        externalObjectDirectoryEntity.setTransferAttempts(++currentNumberOfAttempts);
        if (currentNumberOfAttempts > armDataManagementConfiguration.getMaxRetryAttempts()) {
            logApi.armPushFailed(externalObjectDirectoryEntity.getId());
        }
```